api/base/exceptions.py

```diff
@@ -80,20 +80,82 @@ class Conflict(APIException):
     default_detail = ('Resource identifier does not match server endpoint.')
 
 
-class InvalidQueryStringError(JSONAPIException):
+class JSONAPIParameterException(JSONAPIException):
+    def __init__(self, detail=None, parameter=None):
+        source = {
+            'parameter': parameter
+        }
+        super(JSONAPIParameterException, self).__init__(detail=detail, source=source)
+
+
+class JSONAPIAttributeException(JSONAPIException):
+    def __init__(self, detail=None, attribute=None):
+        source = {
+            'pointer': '/data/attributes/{}'.format(attribute)
+        }
+        super(JSONAPIAttributeException, self).__init__(detail=detail, source=source)
+
+
+class InvalidQueryStringError(JSONAPIParameterException):
     """Raised when client passes an invalid value to a query string parameter."""
     default_detail = 'Query string contains an invalid value.'
     status_code = http.BAD_REQUEST
 
-    def __init__(self, detail=None, parameter=None):
-        super(InvalidQueryStringError, self).__init__(detail=detail, source={
-            'parameter': parameter
-        })
 
+class InvalidFilterOperator(JSONAPIParameterException):
+    """Raised when client passes an invalid operator to a query param filter."""
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None, value=None):
+        if value and not detail:
+            detail = "Value '{0}' is not a supported filter operator; use one of eq, lt, lte, gt, gte.".format(value)
```

Comment:
Won't the list of supported operators vary depending on the field type? CharFields support `icontains`, correct? Perhaps the list of supported operators should be parameterized.